framework/src/util/Utils.cpp

```diff
@@ -154,6 +154,13 @@ std::string GetCurrentWorkingDirectory()
   return std::string();
 }
 
+static const std::string s_CurrWorkingDir = InitCurrentWorkingDirectory();
```

Comment:
Hm, I am not sure we are talking about the same thing. Some OS API (in this particular case the Windows `GetCurrentDirectory` function) must not be called concurrently. How would a static local or call_once approach prevent clients from calling a framework function which initializes the static local (or calls call_once) while concurrently calling `GetCurrentDirectory` through some other code path? Whenever multiple threads can be created in the process, there is no way for any library to synchronize on OS functions (except for some techniques we really do not want to use I would say).

So static initialization - when the process is effectively in "single-threading-mode" - is the only viable solution as far as I can see. Or not call the function at all and require a framework property at framework initialization.

Regarding library dependencies and static globals, the concerns are valid in case of static linking. However, the POSIX `getcwd` is a syscall on Linux (and possibly others) and the Windows `GetCurrentDirectory` is provided by kernel32.dll and cannot be statically linked. If on some systems the current directory cannot be retrieved during initialization, it can still be set as a framework property externally.